app/javascript/src/components/Navigation/components/GlobalActivitiesModal.jsx

```diff
@@ -60,39 +60,73 @@ type State = {
 class GlobalActivitiesModal extends Component<Props, State> {
   displayActivities: Function;
   addMoreActivities: Function;
+  loadData: Function;
+  onCloseModalActions: Function;
 
   constructor(props: Props) {
     super(props);
-    this.state = { activities: this.props.activites, more: this.props.more };
+    this.state = { activities: [], more: false };
     this.displayActivities = this.displayActivities.bind(this);
     this.addMoreActivities = this.addMoreActivities.bind(this);
+    this.onCloseModalActions = this.onCloseModalActions.bind(this);
+    this.loadData = this.loadData.bind(this);
+  }
+
+  loadData() {
+    getActivities().then(response => {
+      this.setState({
+        activities: response.activities,
+        more: response.more
+      });
+    });
+  }
+
+  onCloseModalActions() {
+    this.setState({ activities: [], more: false });
+    this.props.onCloseModal();
   }
 
   displayActivities() {
     if (this.state.activities.length === 0) {
+      if (this.props.showModal) {
+        this.loadData();
+      }
+
       return (
         <li>
           <FormattedMessage id="activities.no_data" />
         </li>
       );
     }
     return this.state.activities.map((activity, i, arr) => {
+      const dateNow = new Date();
       const newDate = new Date(activity.created_at);
       if (i > 0) {
         const prevDate = new Date(arr[i - 1].created_at);
-        if (prevDate < newDate) {
+        if (prevDate.getDate() > newDate.getDate()) {
           return [
             <ActivityDateElement key={newDate} date={newDate} />,
-            <ActivityElement key={activity.id} activity={activity} />
+            <ActivityElement key={i} activity={activity} />
```

Comment:
It's a bad idea to use the array index since it doesn't uniquely identify your elements. Why you are not using `activity.id` anymore?